rpmio/rpmpgp_internal.c

```diff
@@ -572,28 +572,28 @@ static int pgpPrtPubkeyParams(uint8_t pubkey_algo,
 		const uint8_t *p, const uint8_t *h, size_t hlen,
 		pgpDigParams keyp)
 {
-    int rc = 1;
+    int rc = 1; /* assume failure */
     const uint8_t *pend = h + hlen;
     int curve = 0;
+    if (keyp->tag != PGPTAG_PUBLIC_KEY && keyp->tag != PGPTAG_PUBLIC_SUBKEY)
+	return rc; /* Not a public key */
+    if (keyp->alg)
+	return rc; /* We can't handle more than one key at a time */
```

Comment:
This does deserve the comment, but just leave it above the if, like it originally was.